Don't try to parse the pidfile name as a port number.  RT#6750

diff -up krb5/src/kdc/main.c krb5/src/kdc/main.c
--- krb5/src/kdc/main.c	2010-07-07 14:18:30.000000000 -0400
+++ krb5/src/kdc/main.c	2010-07-07 14:18:32.000000000 -0400
@@ -728,6 +728,7 @@ initialize_realms(krb5_context kcontext,
             break;
         case 'P':
             pid_file = optarg;
+            break;
         case 'p':
             if (default_udp_ports)
                 free(default_udp_ports);
